packages/@react-aria/interactions/test/usePress.test.js

```diff
@@ -2992,4 +2993,90 @@ describe('usePress', function () {
       });
     });
   });
+
+  describe('Owner document and window', () => {
+    installPointerEvent();
+
+    const IframeExample = (props) => {
+      React.useEffect(() => {
+        const iframe = document.createElement('iframe');
+        window.document.body.appendChild(iframe);
+        const iframeRoot = iframe.contentWindow.document.createElement('div');
+        iframe.contentWindow.document.body.appendChild(iframeRoot);
+        ReactDOMRender(<Example
+          {...props}
+          data-testid="example" />, iframeRoot);
+      }, [props]);
+
+      return null;
+    };
+
+    test('should receive and handle press events in the correct document and window object', async () => {
+      let events = [];
+      let addEvent = (e) => events.push(e);
+      render(
+        <IframeExample
+          onPressStart={addEvent}
+          onPressEnd={addEvent}
+          onPressChange={pressed => addEvent({type: 'presschange', pressed})}
+          onPress={addEvent}
+          onPressUp={addEvent} />
+      );
+
+      await waitFor(() => {
+        expect(document.querySelector('iframe').contentWindow.document.body.querySelector('div[data-testid="example"]')).toBeTruthy();
+      });
+
+      const el = document.querySelector('iframe').contentWindow.document.body.querySelector('div[data-testid="example"]');
+      fireEvent(el, pointerEvent('pointerdown', {pointerId: 1, pointerType: 'mouse'}));
+      fireEvent(el, pointerEvent('pointerup', {pointerId: 1, pointerType: 'mouse', clientX: 0, clientY: 0}));
```

Comment:
for completeness, we should fire the `click` event as well
don't need to worry about mouse, `usePress` will cancel compatibility the events, but click doesn't get canceled as a result

Reply:
Gotcha!